stdlib/ballerina-builtin/src/main/ballerina/ballerina/collections/vector.bal

```diff
@@ -0,0 +1,147 @@
+package ballerina.collections;
+
+@Description { value: "Vector is a resizable collection type which provides list operations."}
+public struct Vector {
+    // TODO: Make these private once private struct field support is available. Also, make use of init to initialize this
+    any[] vec;
+    int size = 0;
+}
+
+@Description { value: "An error which is returned when the user attempts to access an element which is out of the Vector's range."}
+@Field { value : "msg: The error message"}
+@Field { value : "cause: The cause for the error"}
+@Field { value : "stackTrace: The stack trace of the error"}
+public struct IndexOutOfRangeError {
```

Comment:
Remove this error definition. Is this used anywhere?

Reply:
Yes, this is the error thrown when the user tries to access the vector with an out of bounds index